File: frontend/src/services/attendanceService.js
import axios from "axios";

const API_URL = "https://multimodal-smart-attendance-system-production.up.railway.app/attendance";

export const getAttendance = () => axios.get(API_URL);

export const getAttendanceById = (id) =>
  axios.get(`${API_URL}/${id}`);

export const getAttendanceByDate = (date) =>
  axios.get(`${API_URL}/date/${date}`);

export const getAttendanceByStudent = (studentId) =>
  axios.get(`${API_URL}/student/${studentId}`);

export const addAttendance = (attendance) =>
  axios.post(API_URL, attendance);

export const updateAttendance = (id, attendance) =>
  axios.put(`${API_URL}/${id}`, attendance);

export const deleteAttendance = (id) =>
  axios.delete(`${API_URL}/${id}`);

File: backend/src/main/java/com/attendance/service/AttendanceService.java
package com.attendance.service;

import com.attendance.entity.Attendance;
import com.attendance.repository.AttendanceRepository;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.stereotype.Service;

import java.time.LocalDate;
import java.util.List;

@Service
public class AttendanceService {

    @Autowired
    private AttendanceRepository attendanceRepository;

    // Add Attendance
    public Attendance addAttendance(Attendance attendance) {
        return attendanceRepository.save(attendance);
    }

    // Get All Attendance
    public List<Attendance> getAllAttendance() {
        return attendanceRepository.findAll();
    }

    // Get Attendance By ID
    public Attendance getAttendanceById(Long id) {
        return attendanceRepository.findById(id).orElse(null);
    }

    // Get Attendance By Date
    public List<Attendance> getAttendanceByDate(LocalDate date) {
        return attendanceRepository.findByDate(date);
    }

    // Get Attendance By Student ID
    public List<Attendance> getAttendanceByStudentId(Long studentId) {
        return attendanceRepository.findByStudentId(studentId);
    }

    // Update Attendance
    public Attendance updateAttendance(Attendance attendance) {
        return attendanceRepository.save(attendance);
    }

    // Delete Attendance
    public void deleteAttendance(Long id) {
        attendanceRepository.deleteById(id);
    }
}